lib/cluster/security_group_finder.rb

```diff
@@ -7,9 +7,10 @@ def initialize(vpc)
     end
 
     def find(name)
-      self.class.ec2_client.describe_security_groups.inject([]){ |memo, page| memo + page.security_groups }.find do |group|
+      vpc_filter = { filters: [{ name: "vpc-id", values: [vpc.vpc_id] }] }
+      self.class.ec2_client.describe_security_groups(vpc_filter).inject([]){ |memo, page| memo + page.security_groups }.find do |group|
         # This is tightly coupled to the implementation in templates/OpsWorksinVPC.template
-        group.group_name.match(/#{name}/) && group.vpc_id == vpc.vpc_id
```

Comment:
I see vpc_filter(.., vpc.vpc_id) and ( && group.vpc_id == vpc.vpc_id ).
Don't they do the same thing?

Reply:
the change is to put the vpc filter in the api call so the response only contains security groups relevant to the current cluster. before the change the response included **all** security groups in the entire account and **then** the result was filtered.